### DISCUSSION TITLE
Uploading Files to WhatsApp - Resumable

### DISCUSSION THREAD

**Discussion by theCJMan** [gabrieldwight/Whatsapp-Business-Cloud-Api-Net#68] [Q&A] [answered]
Hi there
I looked at your sample code for File uploads in particular HomeController.UploadMedia using the Resumable method.

However, it seems when I run the code, that I do not get back a WhatsApp media file ID, I get a H from WhatsApp, File Upload Status, and File Offset...

Should i be getting an ID as well so I can use the ID in later sends where I want to refer / use that uploaded file?

The Normal Upload method is working 100%

**Comment by gabrieldwight** [comment]
Resumable upload is only used for interruptions experienced in uploading large files to WhatsApp, it returns the H value which can be used for any graph API that supports file handle requests.

Normal upload returns the media ID that can be used to send any media message with the media ID from the uploaded file.

**Comment by theCJMan** [comment]
Yes, but the point being from your code example, if i had to upload a large file (resumable) then from your code example I'm not getting an ID from WhatsApp.

It sounds like you're saying one then needs to use something in addition to get the file ID from WhatsApp?



⁣Get BlueMail for Android ​

**Reply by gabrieldwight** [reply]
Resumable upload is a global graph API implementation not tied to WhatsApp cloud API. It is aimed at using file handler requests that are used by other graph API services integration like Instagram and Facebook pages that support the file handler requests. The H value is a file handler request and it won't return the media ID from WhatsApp cloud API. Currently, there's no WhatsApp cloud API that supports file handler requests.

For WhatsApp Cloud API only use normal upload for sending media messages with Media ID.

**Comment by theCJMan** [comment]
Ok, now i understand.
So for now, from what you said, the resumable upload should not even be part of this project as WhatsApp cannot use it.

Thank you

⁣Get BlueMail for Android ​

**Reply by gabrieldwight** [reply, marked as answer]
Yes, WhatsApp cannot use resumable upload media to send media messages with file handlers.

**Comment by dedarritchon** [comment]
Hi @gabrieldwight , what about this WhatsApp Cloud API documentations to upload Media on Media Headers Templates or updating the profile pic of a WhatsApp business Profile?

![image](https://github.com/gabrieldwight/Whatsapp-Business-Cloud-Api-Net/assets/17625557/add811d0-42f0-41a6-ae37-5846f03cf74e)
https://developers.facebook.com/docs/whatsapp/business-management-api/message-templates/components

![image](https://github.com/gabrieldwight/Whatsapp-Business-Cloud-Api-Net/assets/17625557/f1cf9e80-d82a-494e-af23-ba1f6394b498)
https://developers.facebook.com/docs/whatsapp/cloud-api/reference/business-profiles/

I'm struggling with these uploads, is there any way to bypas these resumable upload steps? Maybe using an older version of WPP Cloud API?

Best regards,

Daniel

**Reply by dedarritchon** [reply]
The approach of uploading the image directly with the profile_picture_url field is not working, the update returns a success response, but when getting the business profile it comes without profile pic:

`{
    "severity": "INFO",
    "message": "HTTP_CLIENT_POST_REQUEST",
    "commit": "local",
    "url": "https://graph.facebook.com/v18.0/209556495572852/whatsapp_business_profile",
    "headers": {
        "Content-Type": "application/json",
        "Authorization": "Bearer EAAKBNR9DS...ArJg"
    },
    "json": {
        "about": "Adereso BSP",
        "address": "P sherman calle wallaby 42 sydney",
        "description": "Esta es una cuenta para hacer pruebas de Adereso BSP, H\u00e1blame!",
        "email": "contacto@adere.so",
        "messaging_product": "whatsapp",
        "profile_picture_url": "https://media-scl2-1.cdn.whatsapp.net/v/t61.24694-24/319747565_2379646672211515_9106496750750120909_n.jpg?ccb=11-4&oh=01_AdQgsrFuFPaTRK2f2OEjqLHmBXbob0Q-Z0DsOqpDQKubZw&oe=65F57106&_nc_sid=e6ed6c&_nc_cat=108",
        "websites": [
            "https://adere.so/",
            "http://adereso.ai/"
        ],
        "vertical": "PROF_SERVICES"
    }
}

Update business profile response:  {
    "success": true
}

{
    "severity": "INFO",
    "message": "HTTP_CLIENT_GET_REQUEST",
    "commit": "local",
    "url": "https://graph.facebook.com/v18.0/209556495572852/whatsapp_business_profile?fields=about,address,description,email,profile_picture_url,websites,vertical"
}

Get business profile response:  {
    "data": [
        {
            "about": "Adereso BSP",
            "address": "P sherman calle wallaby 42 sydney",
            "description": "Esta es una cuenta para hacer pruebas de Adereso BSP, H\u00e1blame!",
            "email": "contacto@adere.so",
            "websites": [
                "https://adere.so/",
                "http://adereso.ai/"
            ],
            "vertical": "PROF_SERVICES",
            "messaging_product": "whatsapp"
        }
    ]
}
`

**Reply by gabrieldwight** [reply]
Hi, @dedarritchon the whatsapp business profile image was set to use a resumable upload file handle since version 15 of graph API. I think the profile_picture_url should contain the result from the resumable upload with starts with an 'H' string
